src/write_gcm.c

```diff
@@ -2108,7 +2108,7 @@ static char *wg_get_from_aws_metadata_server(const char *resource);
 // Fetches a resource (defined by the concatenation of 'base' and 'resource')
 // from an AWS or GCE metadata server and returns it. Returns NULL upon error.
 static char *wg_get_from_metadata_server(const char *base, const char *resource,
-    const char **headers, int num_headers);
+    const char **headers, int num_headers, const char *provider);
```

Comment:
The only place where it matters is actually `detect_cloud_provider` -- once that's done, you know exactly which one you're querying. Would it make sense to instead add a "ignore_failure" parameter to all three of `wg_get_from_metadata_server`, `wg_get_from_aws_metadata_server`, and `wg_get_from_gcp_metadata_server`?

Reply:
I like the idea! PTAL.